llvm-general/src/LLVM/General/Internal/Attribute.hs

```diff
@@ -50,15 +51,27 @@ instance Monad m => EncodeM m A.PA.ParameterAttribute (Ptr FFI.ParameterAttrBuil
       A.PA.InAlloca -> FFI.parameterAttributeKindInAlloca
       A.PA.NonNull -> FFI.parameterAttributeKindNonNull
       A.PA.Returned -> FFI.parameterAttributeKindReturned
+      A.PA.SwiftSelf -> FFI.parameterAttributeKindSwiftSelf
+      A.PA.SwiftError -> FFI.parameterAttributeKindSwiftError
+      A.PA.WriteOnly -> FFI.parameterAttributeKindWriteOnly
+      A.PA.Alignment _ -> error $ "llvm-general internal error: cases inconsistent in ParameterAttribute encoding for " ++ show a
+      A.PA.Dereferenceable _ -> error $ "llvm-general internal error: cases inconsistent in ParameterAttribute encoding for " ++ show a
+      A.PA.DereferenceableOrNull _ -> error $ "llvm-general internal error: cases inconsistent in ParameterAttribute encoding for " ++ show a
```

Comment:
Please factor these long error messages into a helper function (shared with the other cases, below).